test/e2e/branding_test.go

```diff
@@ -0,0 +1,168 @@
+
+	operatorsv1 "github.com/openshift/api/operator/v1"
+	consoleapi "github.com/openshift/console-operator/pkg/api"
+	"github.com/openshift/console-operator/pkg/testframework"
+)
+
+var validBrands = []operatorsv1.Brand{operatorsv1.BrandOKD, operatorsv1.BrandOCP, operatorsv1.BrandOnline, operatorsv1.BrandDedicated, operatorsv1.BrandAzure}
+
+// Test prep - setup the client used by each test
+func setupBrandingTestCase(t *testing.T) *testframework.Clientset {
+	client := testframework.MustNewClientset(t, nil)
+	return client
+}
+
+// Test the customization-branding features via direct access to the operator config
+func TestOperatorConfigBranding(t *testing.T) {
+	client := setupBrandingTestCase(t)
+	// Save the original config and restore after test
+	originalConfig, err := client.Consoles().Get(consoleapi.ConfigResourceName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("could not get operator config, %v", err)
+	}
+	defer client.Consoles().Update(originalConfig)
+
+	for _, expectedBrand := range validBrands {
+		// helper to update the operator config
+		setOperaterConfigBrand(t, client, expectedBrand)
+		// now check if it has set the brand
+		err = wait.Poll(1*time.Second, pollTimeout, func() (stop bool, err error) {
+			gotBrand := getOperaterConfigBrand(t, client)
+			return (gotBrand == expectedBrand), nil
+		})
+	}
+}
+
+// Test setting brand via the config map in the openshift-config-managed namespace
+// The config map in that namespace is merged with the existing console config map
+func TestBrandingFromManagedConfigMap(t *testing.T) {
+	client := setupBrandingTestCase(t)
+	// Keep original config map so we can reset it after test
+	originalManagedConfigMap, err := client.ConfigMaps(consoleapi.OpenShiftConfigManagedNamespace).Get(consoleapi.OpenShiftConsoleConfigMapName, metav1.GetOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		t.Fatalf("could not get console-config configmap, %v", err)
+	}
+	defer client.ConfigMaps(consoleapi.OpenShiftConfigManagedNamespace).Update(originalManagedConfigMap)
+
+	for _, expectedBrand := range validBrands {
+		t.Logf("update data for the config map in openshift-config-managed namespace with %v", expectedBrand)
+		// Create new configmap for test
+		cm := &v1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "console-config",
+				Namespace: "openshift-config-managed",
+			},
+			Data: map[string]string{
+				"console-config.yaml": `kind: ConsoleConfig
+apiVersion: console.openshift.io/v1
+customization:
+  branding: ` + string(expectedBrand) + `
+  documentationBaseURL: https://docs.okd.io/4.0/
+`,
+			},
+		}
+		_, err = updateOrCreateConsoleConfigMap(client, consoleapi.OpenShiftConfigManagedNamespace, cm)
+		if err != nil {
+			t.Fatalf("error: could not apply managed config map %v", err)
+		}
+
+		err = wait.Poll(1*time.Second, pollTimeout, func() (stop bool, err error) {
+			gotBrand := getConsoleBrand(t, client)
+			return (gotBrand == expectedBrand), nil
+		})
+	}
+}
+
```

Comment:
Test for both?  What if both managed config and operator config are set?  Which one wins?

Reply:
From the console's perspective, the configmap for console comes from the merge that happens in the  configmap->DefaultConfigMap, so the `openshift-config-managed` will win.

Reply:
Right, that is incorrect actually.

Reply:
A unit test might be sufficient, however (once fixed).

Reply:
The configmap in managed is internal. The operator config is an API intended to be used (and have an effect always when set).

Reply:
Right, the config map in openshift-config-managed is just the default when no value is specified. The value in the operator config should win otherwise.

Reply:
I put this on hold until the changes in PR228 merge since this test does work correctly for the current code. PR228 will fix the precedence order for the various configs.